### DISCUSSION TITLE
Jump2D: Global config for `single_character` or `query` built-ins?

### DISCUSSION THREAD

**Discussion by pkazmier** [nvim-mini/mini.nvim#768] [Q&A] [answered]
While testing all of the various built-ins today, when I configure Jump2D to use `single_character` or the `query` built-in, it always states that there are "no spots to show". But, if I invoke `start` via the command prompt, it works as expected:

 `:lua MiniJump2d.start(MiniJump2d.builtin_opts.single_character)`

Here is my setup function where I am trying to set `single_character` as my global setting:

```lua
require('mini.jump2d').setup(require('mini.jump2d').builtin_opts.single_character)
```

While the above does not work as expected, if I use `word_start` on `line_start` as my global setting, it does:

```lua
require('mini.jump2d').setup(require('mini.jump2d').builtin_opts.word_start)
```

What am I doing wrong here? Thank you!

**Comment by echasnovski** [comment, marked as answer]
The `single_character` jumper is a bit of an afterthought, to be honest, hence it is a bit hacky. It **needs** to be a part of `start()` call to work, as [all `builtin_opts` are generally designed to do](https://github.com/echasnovski/mini.nvim/blob/d23e19887170c5a0e780ac98e2649a209ce3e6a8/doc/mini-jump2d.txt#L408-L409). The reason is too technical, but it is what it is.

So for this particular case I suggest creating a separate mapping for it directly. As [described in help page](https://github.com/echasnovski/mini.nvim/blob/d23e19887170c5a0e780ac98e2649a209ce3e6a8/doc/mini-jump2d.txt#L417-L420).

**Reply by pkazmier** [reply]
When you say "hacky", is this because it cannot be used in operator mode? Or am I doing something incorrectly? My lazy key section:

```lua
    keys = {
      { "<cr>", "<Cmd>lua MiniJump2d.start(MiniJump2d.builtin_opts.single_character)<CR>", desc = "Jump anywhere" },
    },
```

**Reply by echasnovski** [reply]
> When you say "hacky", is this because it cannot be used in operator mode?

Not quite. It means that this particular functionality (computing spots based on the single character user input) is implemented in a not the best way (hence not working in some situations when other builtin options do work).

In order for mapping to work in Operator-pending, there should be a mapping created in Operator-pending mode. I don't really know how 'lazy.nvim' managed this, but it seems it does not create such mapping, as I'd assume it means only in Normal mode.

The [suggested earlier approach](https://github.com/echasnovski/mini.nvim/blob/d23e19887170c5a0e780ac98e2649a209ce3e6a8/doc/mini-jump2d.txt#L417-L420) is to create those mappings manually. If you want it to work in all Normal, Visual, and Operator-pending mode, use `{ 'n', 'x', 'o' }` instead of `'n'` when specifying mode in `vim.keymap.set()`.
